test/functional/tests/stats/test_display_statistics.py

```diff
@@ -0,0 +1,420 @@
+
+    with TestRun.step(
+        f"Start {caches_count} caches (one for every cache mode) "
+        f"and add {cores_per_cache} cores per cache"
+    ):
+        caches, cores = cache_prepare(cache_dev, core_dev)
+
+    with TestRun.step(f"Get {stat_filter} statistics for each cache and validate it"):
+        caches_stats = [
+            caches[i].get_statistics_flat(stat_filter=[stat_filter])
+            for i in range(caches_count)
+        ]
+        failed_stats = ""
+        for i in range(caches_count):
+            failed_stats += validate_statistics_flat(
+                caches[i], caches_stats[i], stat_filter, per_core=False
+            )
+
+        if failed_stats:
+            TestRun.LOGGER.error(
+                f"There are some inconsistencies in cache statistics:\n{failed_stats}")
+
+
+@pytest.mark.parametrize(
+    "stat_filter",
+    [StatsFilter.usage, StatsFilter.req, StatsFilter.blk, StatsFilter.err],
+)
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_core_other_stats(stat_filter):
+    """
+        title: Test CAS statistics for core device.
+        description: |
+          Check CAS ability to display usage, request, block and error
+          statistics for core device in every cache mode.
+        pass_criteria:
+          - All core statistics can be retrieved.
+          - No additional statistics are found.
+    """
+
+    with TestRun.step("Partition cache and core devices"):
+        cache_dev, core_dev = storage_prepare()
+
+    with TestRun.step(
+        f"Start {caches_count} caches (one for every cache mode) "
+        f"and add {cores_per_cache} cores per cache"
+    ):
+        caches, cores = cache_prepare(cache_dev, core_dev)
+
+    with TestRun.step(f"Get {stat_filter} statistics for each core and validate it"):
+        failed_stats = ""
+        for i in range(caches_count):
+            cores_stats = [
+                cores[i][j].get_statistics_flat(stat_filter=[stat_filter])
+                for j in range(cores_per_cache)
+            ]
+            for j in range(cores_per_cache):
+                failed_stats += validate_statistics_flat(
+                    cores[i][j], cores_stats[j], stat_filter, per_core=True
+                )
+
+        if failed_stats:
+            TestRun.LOGGER.error(
+                f"There are some inconsistencies in core statistics:\n{failed_stats}")
+
+
+def storage_prepare():
+    cache_dev = TestRun.disks["cache"]
+    cache_parts = [Size(20, Unit.GibiByte)] * caches_count
+    cache_dev.create_partitions(cache_parts)
+    core_dev = TestRun.disks["core"]
+    core_parts = [Size(10, Unit.GibiByte)] * cores_per_cache * caches_count
+    core_dev.create_partitions(core_parts)
+
+    return cache_dev, core_dev
+
+
+def cache_prepare(cache_dev, core_dev):
+    caches = []
+    i = 0
+    for cache_mode in CacheMode:
```

Comment:
for i, cache_mode in enumerate(CacheMode):

Reply:
Done